### DISCUSSION TITLE
Null-Coalescing Operator in C#

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#118] [Q&A] [answered]
Does Objeck support it already or have any plans to support it?

https://www.geeksforgeeks.org/null-coalescing-operator-in-c-sharp/

I'm making Objeck lexer for Lite XL. My lexer is derived from C# lexer. This is left-over of the C# lexer. If Objeck currently not and have no plans to support it in the future I will just remove this line on the lexer:

`{ pattern = "%?%?",                     type = "operator" }, -- ?? null-coalescing`

**Comment by objeck** [comment, marked as answer]
Nope, worth adding? It looks like a ternary shortcut, something I would not personally use in my code.

**Reply by ghost** [reply]
I don't know. I don't use both C# and Objeck. I'm a volunteer making lexers people requested. I'm working on the Objeck lexer for Lite XL. The pattern for null-coalescing is a left-over of C# (the Objeck lexer is based on C# lexer). If you have no plans to support it I will remove the rule for null-coalescing on the lexer.

**Reply by objeck** [reply]
Ok, thank you
